anago

```diff
@@ -620,9 +620,17 @@ prepare_tree () {
     logrun -s git commit -am "Delete extraneous CHANGELOG-*.md files on branch." || return 1
   fi
 
+  # This appears to be a bug fix for some historical bug.  Unclear if/why
+  # this would ever result in label != label_common.
   # Ensure a common name for label in case we're using the special beta indexes
   [[ "$label" =~ ^beta ]] && label_common="beta"
 
+  # Ensure any beta tag is on a unique commit by making a commit here:
+  if [[ "$label_common" == "beta" ]]; then
```

Comment:
should we also verify the commit message at HEAD is not already `"${RELEASE_VERSION[$label]}"`? I don't have context to know if a failure later in this script could result in a rerun that would add another empty commit

Reply:
It should not be a possible scenario for the commit to get added and the associated tag not be added, and that result to be pushed to the repo.

Reply:
ok, and this resets from canonical source on every run (doesn't start with messy local results)?

Reply:
That's our understanding yes.

Reply:
In both `build_tree` (https://github.com/kubernetes/release/blob/c965d4e6ed94eb8cd1ede5fede86d06d7883c5f4/anago#L670) and `prepare_tree` (https://github.com/kubernetes/release/blob/c965d4e6ed94eb8cd1ede5fede86d06d7883c5f4/anago#L596-L604) we do checks.

Reply:
Although at times we've doubted it, anago starts with checking out from upstream master and some checks of the repo state.

Reply:
I think I'd prefer to see `git reset --hard "${RELEASE_VERSION[$label]}"` here, but I'm unsure (at the moment) what else that would affect.

Reply:
I'm not sure what you're wanting that to do... until we tag, `"${RELEASE_VERSION[$label]}"` is not something we can reset to